CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+Thank you for your interest in contributing to the Rust Reference!
+
+There are a few ways of helping with the reference: critiquing the reference,
+editing the reference, fixing incorrect information, adding examples and
+glossary entries, and documenting new or otherwise undocumented features in
+Rust.
+
+For awhile, the Reference was basically ignored, and Rust continued gaining
+new features or changing old ones. It was also basically the introduction
+document before the first edition of the Rust book, and constantly in flux from
+the huge churn of the language design before v1.0.0. So there's a lot that's
+wrong, too teachy for people who should have basic understanding of Rust, or
+is too shallow so as to not go into too much detail in that past. As such, we
+have the warning saying there's work that needs to be done. Eventually, we plan
+to make sure everything is well documented enough that we can remove the
+warning.
+
+## Critiquing the Reference
+
+This is the easiest way to contribute. Basically, as you read the reference, if
+you find something confusing, incorrect, or missing, then you can file an issue
+against the reference explaining your concerns.
+
+## Editing the Reference
+
+Typos and incorrect links get through from time to time. Should you find them,
+we welcome PRs to fix them. Additionally, larger editing jobs that help remove
+the number of parentheticals, remove comma splices, italicize term definitions
+and other similar tasks are helpful.
+
+## Adding Examples and Glossary Entries
+
+Examples are great. Many people will only read examples and ignore the prose.
+Ideally, every facet of every feature will have an example.
+
+Likewise, the reference has a glossary. It doesn't need to explain every facet
+of every feature nor contain every definition, but it does need to be expanded
+upon. Ideally entries in the glossary link to the associated documentation.
+
+## Adding Documentation
+
+There are a lot of features that are not documented at all or are documented
+poorly. This is the hardest, but definitely most valuable. Pick something from
+[Document all features][9] or a [missing feature] tag, and write about it.
+
+While writing, you may find it handy to have a [playpen] open to test out
+
+Feel free to take information from the standard library and Rustonomicon as
+appropriate.
+
+Note that we don't write documentation for purely library features such as
+threads and IO and we don't write about Rust in the future. Documentation is
+written as if the current stable release of Rust is the last release. If you
```

Comment:
Possibly add a note that this isn't entirely true when parts of a feature are stabilized? I'm thinking in particular about `CoerceUnsized`, which isn't stabilized, but where stable code can take advantage of the instances already in `std`.

Reply:
Even in situations like this, I'd rather not mention the not yet stabilized parts, unless it really does help the flow. Your PR does technically break this, and I've been trying to find the right words to say about it. I think it's fine for now, as it is strictly better than what was there before, and at the end of the day, these are all guidelines, not strict policies.